Xamarin.Forms.Platform.iOS/Renderers/ShellSectionRootRenderer.cs

```diff
@@ -296,49 +301,102 @@ protected virtual void OnShellSectionPropertyChanged(object sender, PropertyChan
 					return;
 
 				var currentRenderer = _renderers[newContent];
+				_isAnimatingOut = oldRenderer;
+				_pageAnimation?.StopAnimation(true);
+				_pageAnimation = null;
+				_pageAnimation = CreateContentAnimator(oldRenderer, currentRenderer, oldIndex, newIndex, _containerArea);
 
-				// -1 == slide left, 1 ==  slide right
-				int motionDirection = newIndex > oldIndex ? -1 : 1;
+				if (_pageAnimation != null)
+				{
+					_pageAnimation.AddCompletion((p) =>
+					{
+						if (_isDisposed)
+							return;
 
-				_containerArea.AddSubview(currentRenderer.NativeView);
+						if (p == UIViewAnimatingPosition.End)
+						{
+							RemoveNonVisibleRenderers();
+						}
+					});
 
-				_isAnimatingOut = oldRenderer;
+					_pageAnimation.StartAnimation();
+				}
+				else
+				{
+					RemoveNonVisibleRenderers();
+				}
+			}
+		}
 
-				currentRenderer.NativeView.Frame = new CGRect(-motionDirection * View.Bounds.Width, 0, View.Bounds.Width, View.Bounds.Height);
+		UIViewPropertyAnimator CreateContentAnimator(
```

Comment:
@GiampaoloGabba I'm thinking after this PR is merged we can make this or something close to this virtual for some added animation customizations!!

Reply:
nice! i need to explore the other platforms renderers for the equivalent animation.
For now i jsut played with the NavigationPage (and shell equivalent) animatins, i skipped tabbed because we dont have animations in the TabbedPage for iOS.

For MAUI i will focus more on shell: is more flexibile. As today i dont use much shell because is not well integrated with libraries like PRISM, but from my understating this will change for MAUI, so maybe i will go "full-in" with shell animations